src/v3/src/util/idpButtonIconMap.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import adobeIconSvg from '../img/socialButtonIcons/adobe.svg';
+import amazonIconSvg from '../img/socialButtonIcons/amazon.svg';
+import appleIconSvg from '../img/socialButtonIcons/apple.svg';
+import discordIconSvg from '../img/socialButtonIcons/discord.svg';
+import facebookIconSvg from '../img/socialButtonIcons/facebook.svg';
+import githubIconSvg from '../img/socialButtonIcons/github.svg';
+import gitlabIconSvg from '../img/socialButtonIcons/gitlab.svg';
+import googleIconSvg from '../img/socialButtonIcons/google.svg';
+import lineIconSvg from '../img/socialButtonIcons/line.svg';
+import linkedinIconSvg from '../img/socialButtonIcons/linkedin.svg';
+import msIconSvg from '../img/socialButtonIcons/ms.svg';
+import oktaIconSvg from '../img/socialButtonIcons/okta.svg';
+import orcidIconSvg from '../img/socialButtonIcons/orcid.svg';
+import paypalIconSvg from '../img/socialButtonIcons/paypal.svg';
+import quickbooksIconSvg from '../img/socialButtonIcons/quickbooks.svg';
+import salesforceIconSvg from '../img/socialButtonIcons/salesforce.svg';
+import spotifyIconSvg from '../img/socialButtonIcons/spotify.svg';
+import xeroIconSvg from '../img/socialButtonIcons/xero.svg';
+import yahooIconSvg from '../img/socialButtonIcons/yahoo.svg';
+import yahooJapanIconSvg from '../img/socialButtonIcons/yahoo-japan.svg';
+
+export const idpButtonIconMap: Record<string, string> = {
```

Comment:
Is there any way to type the Record return type so the key is an enum based on the supported IDPs in `src/util/IDP.js`?

Reply:
Seems okay to not do the work to enforce type safety here as long as we have decent default behavior in the case that the lookup in the map returns `undefined`. If we want a "default icon" (not sure if we'd actually want this) we could add an icon to the map with a key of `''`.

Reply:
I want to add this definitely isn't blocking -- just was curious if we wanted to consider doing it while we're here.

Reply:
looks like default icon is not necessary: https://oktainc.atlassian.net/browse/OKTA-609477?focusedCommentId=3601369

Reply:
and currently we do have default behavior in place to handle if the social icon map lookup return `undefined` (it wont try and render an svg)